tests/fuzzers/bn256/bn256_fuzz.go

```diff
@@ -157,11 +157,35 @@ func FuzzPair(data []byte) int {
 		panic("pairing mismatch: cloudflare/google")
 	}
 
-	cPair, err := bn254.Pair([]bn254.G1Affine{*ps}, []bn254.G2Affine{*ts})
+	// gnark uses a different pairing algorithm than google & cloudflare, so the results might not match up
+	// which is not a problem because of the bilinearity of the pairing.
+
+	return 1
+}
+
+func FuzzPairingCheck(data []byte) int {
+	input := bytes.NewReader(data)
+	pc, pg, ps := getG1Points(input)
+	if pc == nil {
+		return 0
+	}
+	tc, tg, ts := getG2Points(input)
+	if tc == nil {
+		return 0
+	}
+
+	// Pair the two points and ensure they result in the same output
+	clOK := cloudflare.PairingCheck([]*cloudflare.G1{pc}, []*cloudflare.G2{tc})
+	gOK := google.PairingCheck([]*google.G1{pg}, []*google.G2{tg})
+	if clOK != gOK {
+		panic("pairing check mismatch: cloudflare/google")
+	}
+
+	cOK, err := bn254.PairingCheck([]bn254.G1Affine{*ps}, []bn254.G2Affine{*ts})
```

Comment:
```suggestion
	// gnark uses a different pairing algorithm than google & cloudflare, so the results might not match up
	// which is not a problem because of the bilinearity of the pairing.

	// Pair the two points and ensure they result in the same output
	clOK := cloudflare.PairingCheck([]*cloudflare.G1{pc}, []*cloudflare.G2{tc})
	gOK := google.PairingCheck([]*google.G1{pg}, []*google.G2{tg})
	if clOK != gOK {
		panic("pairing check mismatch: cloudflare/google")
	}
	cOK, err := bn254.PairingCheck([]bn254.G1Affine{*ps}, []bn254.G2Affine{*ts})
```